packages/contracts/contracts/L1/Colosseum.sol

```diff
@@ -559,20 +497,18 @@ contract Colosseum is Initializable, ISemver {
     /**
      * @notice Proves that a specific output is invalid using ZKP.
      *         This function can only be called in the READY_TO_PROVE and ASSERTER_TIMEOUT states.
+     *         Note that if zkEVM proof is given, proveFault is verified based on zkEVM, otherwise zkVM.
      *
      * @param _outputIndex Index of the L2 checkpoint output.
      * @param _pos         Position of the last valid segment.
-     * @param _proof       Proof for public input validation.
-     * @param _zkproof     Halo2 proofs composed of points and scalars.
-     *                     See https://zcash.github.io/halo2/design/implementation/proofs.html.
-     * @param _pair        Aggregated multi-opening proofs and public inputs. (Currently only 2 public inputs)
+     * @param _zkEVMProof  The public input and proof using zkEVM.
+     * @param _zkVMProof   The public input and proof using zkVM.
      */
     function proveFault(
         uint256 _outputIndex,
         uint256 _pos,
-        Types.PublicInputProof calldata _proof,
-        uint256[] calldata _zkproof,
-        uint256[] calldata _pair
+        Types.ZKEVMProof calldata _zkEVMProof,
+        Types.ZKVMProof calldata _zkVMProof
```

Comment:
Is it impossible to separate the functions `proveFaultZkEVM` and `proveFaultZkVM`?